code/modules/antagonists/revenant/revenant_abilities.dm

```diff
@@ -1,12 +1,20 @@
 
 //Harvest; activated ly clicking the target, will try to drain their essence.
 /mob/living/simple_animal/revenant/ClickOn(atom/A, params) //revenants can't interact with the world directly.
-	A.examine(src)
+	var/list/modifiers = params2list(params)
+	if(modifiers["shift"])
+		ShiftClickOn(A)
+		return
+	if(modifiers["alt"])
+		AltClickOn(A)
```

Comment:
Given how leaky our AltClick checks are this enables a lot of interactions. I'd limit this to working only when revealed.

Reply:
Revenants cannot choose to reveal themselves, they are only revealed when forced to by spells. This is needed to be able to drain corpses that are piled together, or just underneath something. 

I did test a few things like APC's, and revenant being simplemob blocks those interactions at least. Then again, seems you can open/close sleepers so I see what you mean.

One alternative is to completely sidestep the established alt-click procs and just do the turf listing directly in the revenant ClickOn. How do you feel about that?

Reply:
Sure, ideally extract the functionality to some type independent helper and call that in both places.

Reply:
There is one already, but it cannot be used without allowing atoms to override it. I added a new one, specifically for turf listing and nothing more. It needs to be used in a few more places it seems, so I will make a separate PR for it.